conformance/utils/echo/pod.go

```diff
@@ -113,32 +113,27 @@ func compareRequest(exp http.ExpectedResponse, resp Response) error {
 	return nil
 }
 
-// copied from conformance/http to get the ipv6 matching
-func calculateHost(reqHost, scheme string) string {
+func calculateHost(t *testing.T, reqHost, scheme string) string {
 	host, port, err := net.SplitHostPort(reqHost)
+	if err != nil && strings.Contains(err.Error(), "too many colons in address") {
+		// This is an IPv6 address; assume it's valid ipv6
+		// Assume caller won't add a port without brackets
+		reqHost = "[" + reqHost + "]"
+		host, port, err = net.SplitHostPort(reqHost)
+	}
 	if err != nil {
+		t.Logf("Failed to parse host %q: %v", reqHost, err)
```

Comment:
`t.Errorf` or `t.Logf` to print errors ?

Reply:
These aren't fatal errors, nor are they being bubbled up and reported, so I opted for a generic log. Note that the test continues even if an error in parsing occurs. I would choose a warn log level if it existed